js/tree.js

```diff
@@ -372,11 +503,36 @@
 	};
 
 	$.fn.tree.defaults = {
-		dataSource: function (options, callback) {},
+		dataSource: function dataSource(options, callback) {},
 		multiSelect: false,
 		cacheItems: true,
 		folderSelect: true,
-		itemSelect: true
+		itemSelect: true,
+		/*
+		* currently calling "open" again on a folder will close it. Setting
+		* `ignoreRedundantOpens` to `true` will make the folder instead ignore
+		* the redundant call and stay open.
+		*/
+		ignoreRedundantOpens: false,
+		/*
+		* this is not necessarily indicative of how many layers deep the tree goes,
+		* only how many times `discloseAll` should be called before a warning trigger
+		* is fired for the dev to hook into. `discloseAll` will continue running
+		* after this many recursions unles the dev manually kilss it by setting
+		* `keep-disclosing` to `false`:
```

Comment:
Intention here is to help devs protect themselves. The discloseAll is recursive like in that it calls itself after the folder is loaded. If there are many layers in the folder structure this could go on for a long time but the browser would not stop it since it is not true recursion. This allows the dev to hook into the event and decided what to do with the fringe case. `keep-disclosing` is a kill switch.